### PR TITLE
test: ported route 3 test

### PR DESCRIPTION
**Pull request by kriskw1999**
## Description 

Ported to the node test runner: `route.3.test.js`

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by jsumners — APPROVED** [review]
Looks good to me.

File: test/route.3.test.js [hunk 1]
```diff
@@ -32,22 +31,24 @@ test('does not mutate joi schemas', t => {
       params: { an_id: joi.number() }
     },
     handler (req, res) {
-      t.same(req.params, { an_id: 42 })
+      console.log(req.params)
+      t.assert.deepEqual(req.params, { an_id: 42 })
```
Comment (Eomm) [inline test/route.3.test.js:35]:
```suggestion
      t.assert.deepEqual(req.params, { an_id: 42 })
```

File: test/route.3.test.js [hunk 2]
```diff
@@ -32,22 +31,24 @@ test('does not mutate joi schemas', t => {
...
     method: 'GET',
     url: '/foo/42'
-  }, (err, result) => {
-    t.error(err)
-    t.equal(result.statusCode, 200)
-    t.same(JSON.parse(result.payload), { hello: 'world' })
+  }, (err, res) => {
+    t.assert.ifError(err)
+    t.assert.strictEqual(res.statusCode, 200)
+    t.assert.deepStrictEqual(JSON.parse(res.payload), { hello: 'world' })
```
Comment (Eomm) [inline test/route.3.test.js:46]:
```suggestion
    t.assert.deepStrictEqual(res.json(), { hello: 'world' })
```

File: test/route.3.test.js [hunk 3]
```diff
@@ -159,23 +162,24 @@ test('route error handler does not affect other routes', t => {
     method: 'GET',
     url: '/tea'
   }, (error, res) => {
-    t.error(error)
-    t.equal(res.statusCode, 500)
-    t.same(JSON.parse(res.payload), {
+    t.assert.ifError(error)
+    t.assert.strictEqual(res.statusCode, 500)
+    t.assert.deepStrictEqual(JSON.parse(res.payload), {
```
Comment (Eomm) [inline test/route.3.test.js:167]:
```suggestion
    t.assert.deepStrictEqual(res.json(), {
```

File: test/route.3.test.js [hunk 4]
```diff
@@ -197,12 +201,13 @@ test('async error handler for a route', t => {
     method: 'GET',
     url: '/late-coffee'
   }, (error, res) => {
-    t.error(error)
-    t.equal(res.statusCode, 418)
-    t.same(JSON.parse(res.payload), {
+    t.assert.ifError(error)
+    t.assert.strictEqual(res.statusCode, 418)
+    t.assert.deepStrictEqual(JSON.parse(res.payload), {
```
Comment (Eomm) [inline test/route.3.test.js:206]:
```suggestion
    t.assert.deepStrictEqual(res.json(), {
```

**Comment by kriskw1999** [comment]
@Eomm updated with all requested changes, thank you

**Review by Eomm — APPROVED** [review]

**Pull request merged**